### DISCUSSION TITLE
Experimental (alpha) APIs graduation

### DISCUSSION THREAD

**Discussion by evgenyka** [aws/aws-cdk#24038] [Polls]
Hey y'all!
On 08-Nov-21, we [introduced AWS CDK v2 experimental (alpha) APIs](https://aws.amazon.com/blogs/developer/experimental-construct-libraries-are-now-available-in-aws-cdk-v2/) (https://aws.amazon.com/blogs/developer/experimental-construct-libraries-are-now-available-in-aws-cdk-v2/) that are available as a separated packages, in addition to existing stable APIs.
We create a new alpha module the first time L2 constructs are created for a new service or core module. After an alpha module stabilizes, it is moved to aws-cdk-lib and have a guaranteed backwards compatibility within the AWS CDK major version. The alpha modules are denoted with an -alpha identifier, to clearly separate them from a stable modules.
Due to the high demand and pace of new L2 constructs creation, by 01-Jan-23 we have 30 alpha modules.
We do want to use the power of our community and prioritize the work of stabilizing those. Hence, we are looking for your opinion!
Note: the discussion does not represent a commitment, guarantee, obligation, or promise to deliver alpha module stabilization by any particular date.

List of alpha modules (as of 01-Jan-23):

01 | apigatewayv2-alpha
02 | glue-alpha (in progress)
03 | batch-alpha (in progress)
04 | amplify-alpha (in progress)
05 | kinesisfirehose-alpha
06 | redshift-alpha
07 | cognito-identitypool-alpha
08 | servicecatalogappregistry-alpha
09 | synthetics-alpha
10 | neptune-alpha
11 | iot-alpha
12 | kinesisanalytics-flink-alpha
13 | msk-alpha
14 | apprunner-alpha
15 | route53resolver-alpha
16 | ivs-alpha
17 | s3objectlambda-alpha
18 | iotevents-alpha
19 | cloud9-alpha
20 | sagemaker-alpha
21 | location-alpha
22 | codestar-alpha
23 | apigatewayv2-authorizers-alpha
24 | apigatewayv2-integrations-alpha
25 | gamelift-alpha
26 | iot-actions-alpha
27 | iotevents-actions-alpha
28 | kinesisfirehose-destinations-alpha

**Comment by dbartholomae** [comment]
My list would be:
1. amplify-alpha
2. apigatewayv2-alpha

**Reply by TheRealAmazonKendra** [reply]
Amplify is not on the voting list because it is already in progress.

**Comment by gshpychka** [comment]
Other - glue-alpha

**Reply by TheRealAmazonKendra** [reply]
This is already in progress.

**Comment by RichiCoder1** [comment]
Absolutely for `apigatewayv2-*` in general as I think it's a commonly used component.

I'm actually pretty interested in servicecatalogappregistry-alpha, especially with some of the CDK integration working as that + https://aws.amazon.com/blogs/aws/new-amazon-cloudwatch-cross-account-observability/ (which would be awesome to start an L2 for) could open the door for some awesome cross-account CDK-created visibility stuff.

**Comment by johnschultz** [comment]
My vote would be for: `12 | kinesisanalytics-flink-alpha`

We're making more and more use of Flink at Twitch, so it would be nice to have the official construct stabilized.

**Comment by FlorinAsavoaie** [comment]
appsync-alpha

**Reply by piotrmoszkowicz** [reply]
AppSync is already stable ;)

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appsync-readme.html

**Comment by mrpackethead** [comment]
This list is going to be driven by what ever is burning you the most at the time you asnwer. :-) In my case redshift. Next week, might be somethign new.

**Comment by NetDevAutomate** [comment]
Transit Gateway
Cloud WAN / NetworkManager
Network Firewall
Firewall Manager
ALB
NLB
WAF
PrivateLink
Route53

**Comment by mtliendo** [comment]
Are [EventBridge Pipes](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_pipes-readme.html) missing from the list?

**Reply by evgenyka** [reply]
There are no official hand-written ([L2](https://docs.aws.amazon.com/cdk/latest/guide/constructs.html#constructs_lib)) constructs for this service yet.

**Comment by evgenyka** [comment]
Locking the 2023-Q1 conversation - thanks a lot for your votes and comments!
We will create new pool for 2023-Q2 soon.
